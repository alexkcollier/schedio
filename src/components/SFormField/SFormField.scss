.form-field {
  & + & {
    margin-top: spacing('loose');
  }
}
