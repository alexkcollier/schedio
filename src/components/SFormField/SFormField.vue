<template>
  <div class="form-field">
    <slot />
  </div>
</template>

<script>
export default {
  name: 'SFormField'
}
</script>
