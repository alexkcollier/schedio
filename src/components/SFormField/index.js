import SFormField from './SFormField.vue';

export { SFormField };

export default {
  install: Vue => Vue.component(SFormField.name, SFormField),
};
