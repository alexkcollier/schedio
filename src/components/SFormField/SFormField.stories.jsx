import { SFormField } from '@/components/SFormField'
import { SHeading } from '@/components/SHeading'
import PropList from '@/docs/PropList'
import StoryContainer from '@/docs/StoryContainer'
import { withKnobs } from '@storybook/addon-knobs'
import { storiesOf } from '@storybook/vue'

storiesOf('Components/SFormField', module)
  .addDecorator(withKnobs)
  .add('SFormField', () => {
    return {
      render(h) {
        const submitHandler = e => {
          e.preventDefault()
          alert('You submitted the form')
        }

        return (
          <StoryContainer>
            <SHeading level="1">SFormField</SHeading>
            <p>This is a simple wrapper component to group and space form elements.</p>

            <SHeading level="2">Example</SHeading>
            <form onsubmit={submitHandler}>
              <SFormField>
                <SInput id="input" name="input" label="input" required={false} />
              </SFormField>

              <SFormField>
                <SInput id="input-two" name="input-two" label="input-two" required={false} />
              </SFormField>

              <SFormField>
                {['1', '2', '3'].map(r => (
                  <SRadio
                    name="radios"
                    id={`radio-${r}`}
                    label={`radio-${r}`}
                    required={false}
                    showOptional={false}
                    checked={r === '1'}
                  />
                ))}
              </SFormField>

              <SFormField>
                <SButton button-color="green">Submit</SButton>
              </SFormField>
            </form>

            {SFormField.props && <PropList component={SFormField} />}
          </StoryContainer>
        )
      }
    }
  })
