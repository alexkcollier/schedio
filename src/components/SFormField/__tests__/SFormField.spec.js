import { shallowMount } from '@vue/test-utils'
import SFormField from '@/components/SFormField/SFormField.vue'

describe('SFormField.vue', () => {
  it('renders correctly', () => {
    const wrapper = shallowMount(SFormField)

    expect(wrapper.html()).toMatchSnapshot()
  })
})
