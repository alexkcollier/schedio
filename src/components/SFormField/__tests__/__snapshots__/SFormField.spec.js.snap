// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SFormField.vue renders correctly 1`] = `<div class="form-field"></div>`;
